draft-ietf-moq-transport.md

```diff
@@ -954,47 +981,54 @@ to the old relay can be stopped with an UNSUBSCRIBE.
-SUBSCRIBE to.
+SUBSCRIBE to.  Alternately, the publisher can send one or more PUBLISH messages
+to establish subscriptions to tracks.  Relays MAY respond with PUBLISH_OK in
+Forward State=0 until there are known subscribers for new tracks.
 
-Relays MUST verify that publishers are authorized to publish
-the content associated with the set of
-tracks whose Track Namespace matches the announced namespace. Where the
-authorization and identification of the publisher occurs depends on the way the
-relay is managed and is application specific.
+Relays MUST verify that publishers are authorized to publish the content
+associated with the set of tracks whose Track Namespace matches the namespace in
+an ANNOUNCE, or the Full Track Name in PUBLISH. Where the authorization and
+identification of the publisher occurs depends on the way the relay is managed
+and is application specific.
 
-A Relay can receive announcements from multiple publishers for the same
-Track Namespace and it SHOULD respond with the same response to each of the
-publishers, as though it was responding to an ANNOUNCE
-from a single publisher for a given track namespace.
+A Relay can receive announcements for the same Track Namespace or PUBLISH
+messages for the same track from multiple publishers, and it SHOULD respond with
+the same response to each of the publishers.
 
 When a publisher wants to stop new subscriptions for an announced namespace it
 sends an UNANNOUNCE. A subscriber indicates it will no longer subcribe to tracks
 in a namespace it previously responded ANNOUNCE_OK to by sending an
 ANNOUNCE_CANCEL.
 
-A relay manages sessions from multiple publishers and subscribers,
-connecting them based on the track namespace. This MUST use an exact
-match on track namespace unless otherwise negotiated by the application.
-For example, a SUBSCRIBE namespace=foobar message will be forwarded to
-the session that sent ANNOUNCE namespace=foobar.
-
-When a relay receives an incoming SUBSCRIBE request that triggers an
-upstream subscription, it SHOULD send a SUBSCRIBE request to each
-publisher that has announced the subscription's namespace, unless it
-already has an active subscription for the Objects requested by the
-incoming SUBSCRIBE request from all available publishers.
+A relay manages sessions from multiple publishers and subscribers, connecting
+them based on the track namespace or track name. This MUST use an exact match on
+track namespace and name, if present, unless otherwise negotiated by the
+application.  For example, a SUBSCRIBE namespace=foobar, track=x message will be
+forwarded to the session that sent ANNOUNCE namespace=foobar, but not one that
+sent ANNOUNCE namespace=foo.
+
+When a relay receives an incoming SUBSCRIBE request that triggers an upstream
+subscription, it SHOULD send a SUBSCRIBE request to each publisher that has
+announced the subscription's namespace, unless it already has an active
```

Comment:
```suggestion
announced the subscription's namespace or received PUBLISH on the same FullTrackName, unless it already has an active
```

Reply:
If the publisher has established a subscription using PUBLISH then that falls into the second half of the sentence "already has an active subscription" -- both PUBLISH and SUBSCRIBE establish subscriptions.